cornell-dti/course-reviews-react-2.0
# Results Page Redesign and Functionality Implementation

### Summary 

This pull request implements the new search and results page functionality. 

- [x] implemented the redesigned result page
- [x] implemented sorting and filtering on the results page
- [x] handled the "no result" case
- [x] hooked up the results page to the backend Meteor function to retrieve the filtered results

### Test Plan 

Correct PreviewCard clicked
- The first class in the returned list has its PreviewCard component displayed by default and its border is colored to blue to indicate “clicked”
- Clicking another result updates the PreviewCard component to display the new class’ information, changes the border of the new result to blue, and the other result’s border returns to its original color

Sorting
- The returned list is sorted by Overall Rating by default 
- Change the metric to sort by and check that the classes’ new values and colors are sorted in the correct order
- Default preview card is updated to the first class of the sorted list and the first class’ border is highlighted

Filtering
- Each filter individually works
- Combining filters work as expected (e.g. if both semester and class level are checked, only classes at that class level and offered in that semester are displayed)
- Removing filters works as expected
- Results are sorted by the indicated metric even when filtering
- Default preview card is updated to the first class of the filtered list and the first class’ border is highlighted

No Results
- If the list returned from the call is empty, results page is “Sorry…”, and the top says that “0 courses” are returned
- For a non-empty returned list, checking a combination of filters that none of the classes in the list satisfy creates a “Sorry…” page
- Unchecking these filters returns the page to its original state
- Sort the classes by a different metric, check the filters that lead to no results, uncheck these filters and ensure that the sorting remains the same, including the metric selected in the “Sort By”
- Check the filters that lead to no results, change the “Sort By” metric, uncheck these filters and ensure the expected class list is returned and sorted
